pkg/nvcdi/lib.go

```diff
@@ -145,7 +150,7 @@ func New(opts ...Option) (Interface, error) {
 			l.vendor = "management.nvidia.com"
 		}
 		// Management containers in general do not require CUDA Forward compatibility.
-		l.disabledHooks[HookEnableCudaCompat] = true
+		l.hookCreator.DisableHook(string(HookEnableCudaCompat))
```

Comment:
We either need to keep this as is, or properly move the handling of disabled hooks to the `discover` package. I would recommend handling skipping here in the first commit / PR and then adding proper disabled functionality in a follow-up commit or PR.

Reply:
done